plugins/inmemoryscanner/src/lib/Scanner.cpp

```diff
@@ -45,67 +50,56 @@ namespace InMemoryScanner
         if (memoryRegionDescriptor.isSharedMemory && !memoryRegionDescriptor.isProcessBaseImage)
         {
             verdict = false;
-            pluginInterface->logMessage(
-                LogLevel::info, LOG_FILENAME, "Skipping: Is shared memory and not the process base image.");
+            logger->info("Skipping: Is shared memory and not the process base image.");
         }
         return verdict;
     }
 
     void
     Scanner::scanMemoryRegion(pid_t pid, const std::string& processName, const MemoryRegion& memoryRegionDescriptor)
     {
-        pluginInterface->logMessage(LogLevel::info,
-                                    LOG_FILENAME,
-                                    "Scanning Memory region from " + intToHex(memoryRegionDescriptor.base) +
-                                        " with size " + intToHex(memoryRegionDescriptor.size) + " name " +
-                                        memoryRegionDescriptor.moduleName);
+        logger->info("Scanning Memory region",
+                     {{"VA", fmt::format("{:x}", memoryRegionDescriptor.base)},
+                      {"Size", memoryRegionDescriptor.size},
+                      {"Module", memoryRegionDescriptor.moduleName}});
 
         if (shouldRegionBeScanned(memoryRegionDescriptor))
         {
             auto scanSize = memoryRegionDescriptor.size;
             auto maximumScanSize = configuration->getMaximumScanSize();
             if (scanSize > maximumScanSize)
             {
-                pluginInterface->logMessage(
-                    LogLevel::info, LOG_FILENAME, "Memory region is too big, reduce to " + intToHex(maximumScanSize));
+                logger->info("Memory region is too big", {{"Size", scanSize}});
```

Comment:
missing the size reduction logging. With this message one would expect the region to be skipped.